### ISSUE TITLE
URL Issue

### ISSUE THREAD

**Issue by garnern2** [bramkragten/lyric#2]
I keep getting after clicking to authorize:

`{"code":400, "message":"The redirect URL provided does not match the redirect URL registered for the app."}`

I used the same url as my base url, but it's not working. Is there an update that I'm missing?

**Comment by mdezzi** [comment]
I have not had any success with bramkragten's version and my Lyric T5. Try [this version](https://community.home-assistant.io/t/honeywell-lyric-thermostat/3520/567)

**Comment by garnern2** [comment]
Thanks! I got much further with the component you linked. I get to the page where I select my devices (I have 2), and once I click "Connect," I get "This page isn't working." Any ideas?

**Comment by mdezzi** [comment]
Can you send a screenshot?

**Comment by garnern2** [comment]
I mean I can but that’s literally all it says.

**Comment by mdezzi** [comment]
Ok, that's weird i got just about every error after clicking connect but have never seen that one. Here's how i've been able to make it work:
In honeywell developer console, my Callback URL is http://192.168.1.200:8123/api/lyric/authenticate. 
I make sure that i dont use https, and make sure to add the port number. 

Then when i configure the honewell, i make sure i am accessing HA from inside my local network and at the same url (i.e. http://192.168.1.200:8123, not https),

That has been successful a few times. Additionally, if you see a lyric.conf file in your config folder, delete that and try again.

**Comment by niwasgrewal** [comment]
changed the callback from Https:// to Http:// and it will fix the issue.

**Comment by kysdaddy** [comment]
I hope it's cool to jump in here, I am running into a similar issue. When I corrected my callback too; http://192.168.1.2:8123/api/lyric/authenticate I now get 

500 Internal Server Error
Server got itself in trouble

And can't figure out where to go next.

Chas

**Comment by mdezzi** [comment]
Dont bother with the BramKragten version or the one in HACS. 

[this custom component works well](https://github.com/shellster/lyric). I just installed it and connected my T5.

**Comment by kysdaddy** [comment]
I tried every version I could find and they all ended with the 500 internal server error, any ideas?

Chas

**Comment by mdezzi** [comment]
I set that custom component up yesterday, here's what i did.

- copy all the files to the custom_components folder
- set the callback URL in the honeywell developer site is http://IPADDRESS:8123/api/lyric/authenticate. For me it is http://10.0.0.200:8123/api/lyric/authenticate. Don't use https, make sure you have the port. 
- in your config, make sure you have "base_url" set to the same ip address. For me it is:
````
  base_url: http://10.0.0.200:8123
````
- Enter your client key and secret in the config.
- Reboot and make sure you are accessing your server locally via the same IP address, and NOT via https, and not via HA cloud or from outside your network.

**Comment by kysdaddy** [comment]
Thank you.
I have done the same thing several times but continually run into the
500 Internal Server ErrorServer got itself in trouble

Error.

Any idea how to fix this issue?

Chas

**Comment by mdezzi** [comment]
No, i had the same issues with the BramKragten and HACS version (i think they're the same), but the shellster version i linked worked fine. Sorry i dont have any other ideas

**Comment by kysdaddy** [comment]
did you do anything special after removing the previous versions to ensure
that everything was cleared?

**Comment by mdezzi** [comment]
I actually moved into a new house, so i started from scratch.

**Comment by garnern2** [comment]
The @shellster for @mdezzi linked is working very well for me now, even with a Honeywell T9. I'm leaving this open because the original still isn't working, but I want others to see this solution.

**Comment by shellster** [comment]
I believe the changes are now merged into this repo and I'm continuing work on getting an official module into Home Assistant, hopefully later this month or early next.

**Comment by garnern2** [comment]
That's great to hear. Hopefully full function of the T9 is in the works, too. I'd love to use the Honeywell sensors for presence detection.
